pkg/adsc/adsc.go

```diff
@@ -442,10 +442,30 @@ func (a *ADSC) handleRecv() {
 				}
 			}
 		}
+		// last resource of this type. IstioStore doesn't yet have a way to indicate this
+		// or to support large sets - memory store has a special handling by calling Create
+		// with an empty resource. This is used to avoid sleep loop on HasSynced
+		if a.Store != nil && len(gvk) == 3 {
+			_, err = a.Store.Create(model.Config{
```

Comment:
if we are going to make a memory controller specific hack why not just make a proper first class function on memory controller like `ReportSynced()` and stop using the interface? Using `IstioStore` doesn't have much value if you are relying on implementation details of a single store, might as well use concrete types?

Reply:
Yes, that's what I was thinking long term. But not add a function - instead an interface. 

Right now the immediate problem is that memory store can't handle 'synced' - k8s and MCP have no problem with that, and don't need changes. Usually it's best to start with a single store - and evaluate if it needs to be made generic - or part of the model interface, and how. I'm not convinced of either.